### PR TITLE
Further improvements of nn.container docs

### PR DESCRIPTION
**Pull request by ssnl**

### CODE REVIEW

File: torch/nn/modules/container.py [hunk 1]
```diff
@@ -436,9 +441,21 @@ class ParameterDict(Module):
...
 
+    :class:`~torch.nn.ParameterDict` is an **ordered** dictionary that respects
+
+    * the order of insertion, and
+
+    * in :meth:`~torch.nn.ParameterDict.update`, the order of the merged ``OrderedDict``
+      or another :class:`~torch.nn.ParameterDict` (the argument to
+      :meth:`~torch.nn.ParameterDict.update`).
+
+    Note that :meth:`~torch.nn.ParameterDict.update` with other unordered mapping
```
Comment (vishwakftw) [inline torch/nn/modules/container.py]: Could this added in a note-block?

Comment (SsnL) [reply torch/nn/modules/container.py]: It sort of goes together with the previous paragraph, so I didn't make it a .. note::

File: torch/nn/modules/container.py [hunk 2]
```diff
@@ -436,9 +441,21 @@ class ParameterDict(Module):
...
+    types (e.g., Python's plain ``dict``) doesn't not preserve order of the
```
Comment (zou3519) [inline torch/nn/modules/container.py]: doesn't not -> does not

File: torch/nn/modules/container.py [hunk 3]
```diff
@@ -510,21 +527,25 @@ def values(self):
...
+                key-value pairs of type (string, :class:`~torch.nn.Parameter`)
         """
         if not isinstance(parameters, container_abcs.Iterable):
             raise TypeError("ParametersDict.update should be called with an "
                             "iterable of key/value pairs, but got " +
                             type(parameters).__name__)
 
         if isinstance(parameters, container_abcs.Mapping):
-            if isinstance(parameters, OrderedDict):
+            if isinstance(parameters, (OrderedDict, ParameterDict)):
```
Comment (zou3519) [inline torch/nn/modules/container.py]: Interesting, didn't know you could do this

**Review by zou3519 — APPROVED** [review]
One comment on something that looks like a typo. Otherwise, looks good to me

File: torch/nn/modules/container.py (hunk 3 above)
Comment (SsnL) [reply torch/nn/modules/container.py]: yeah it takes in a tuple :D

**Comment by SsnL** [comment]
@zou3519 Thanks! I've fixed the typo. Hopefully lint passes as I edited on github UI directly.

**Pull request closed without merging**